code/game/turfs/simulated/walls.dm

```diff
@@ -82,11 +86,21 @@
 	return "You can deconstruct this by welding it, and then wrenching the girder.<br>\
 			You can build a wall by using metal sheets and making a girder, then adding more metal or plasteel."
 
+/// Apply rust effects to the wall
+/turf/simulated/wall/proc/rust()
+	if(rusted)
+		return
+	name = "rusted [name]"
+	rusted = TRUE
+	update_icon()
+
 /turf/simulated/wall/proc/update_icon()
 	if(!damage_overlays[1]) //list hasn't been populated
 		generate_overlays()
 
 	QUEUE_SMOOTH(src)
+	if(rusted && !rusted_applied)
+		add_overlay(icon('icons/turf/overlays.dmi', pick("rust", "rust2"), pick(NORTH, SOUTH, EAST, WEST)))
```

Comment:
You probably meant
```suggestion
	if(rusted && !rusted_applied)
		add_overlay(icon('icons/turf/overlays.dmi', pick("rust", "rust2"), pick(NORTH, SOUTH, EAST, WEST)))
		rusted_applied = TRUE
```

Reply:
Hmm, I'm not well versed enough in icon operations to know